src/REGothEngine.hpp

```diff
@@ -3,14 +3,69 @@
 
 #pragma once
 
+#include <iostream>
+
 #include <BsPrerequisites.h>
 #include <cxxopts.hpp>
 
+/**
+ * @brief Allows using the bs::Path data type together with cxxopts.
+ * @param str Input stringstream.
+ * @param path Path to write data to.
+ * @return stringstream.
+ */
+std::stringstream& operator>>(std::stringstream& str, bs::Path& path);
+
 namespace REGoth
 {
   class EngineContent;
   class OriginalGameFiles;
 
+  /**
+   * The base configuration of the engine.
+   */
+  class EngineConfig
+  {
+
+  public:
+
+    virtual ~EngineConfig()
+    {
+      // pass
+    }
+
+    void registerCLIEngineOptions(cxxopts::Options& options)
+    {
+      // Configure positional handling
+      options.positional_help("[GAME ASSETS PATH]");
+      options.show_positional_help();
+
+      // Define engine options
+      options.add_options()
+        ("a,game-assets", "Path to a Gothic or Gothic 2 installation", cxxopts::value<bs::Path>(assetsPath), "[PATH]")
+        ("video-x-res", "X resolution", cxxopts::value<unsigned int>(resolutionX), "[PX]")
+        ("video-y-res", "Y resolution", cxxopts::value<unsigned int>(resolutionY), "[PX]")
+        ("video-fullscreen", "Run in fullscreen mode", cxxopts::value<bool>(fullscreen))
+        ;
+
+      // Allow game-assets to also be a positional
+      options.parse_positional({"game-assets"});
+    }
+
+    virtual void registerCLIOptions(cxxopts::Options& /* options */)
+    {
+      // pass
+    }
+
+    unsigned int verbosity = 0;
+    bs::Path gameExecutable;
```

Comment:
Maybe call this so it's obvious that the REGoth executable is meant here? Something like `runningExecutablePath` or similar?

Reply:
Renamed to `engineExecutablePath` (it was called like this in the original code, and I think this is the best fitting. Let me know if you disagree or think `running...` is better :)